chapter07/rwd/ex6-7.pl

```diff
@@ -0,0 +1,17 @@
+#!/urs/bin/perl
+
+use strict;
+use warnings;
+
+if (@ARGV != 1) {
+	die "Error: file name must be supplied.";
+}
+
+while (<>) {
+	if (/fred/) {
+		if (/wilma/) {
```

Comment:
You could also use `&&` rather than nesting the ifs, there shouldn't be any performance difference as if `/fred/` fails then it won't bother to evaluate `/wilma/`.